### ISSUE TITLE
Workbook.getSelectedRange throws error when other Workbook is focused

### ISSUE THREAD

**Issue by ndreisg** [OfficeDev/office-js#561]
**Workbook.getSelectedRange** throws an error when any other Workbook is focused.
** **

## Expected Behavior
Selected Range of the Workbook where the Add-In is running should be returned.

## Current Behavior
Throws the following error:

```
InvalidSelection: The current selection is invalid for this operation.
 at Anonymous function 
  (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:12439:6)
 at lib$es6$promise$$internal$$tryCatch 
  (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:9240:8)
 at lib$es6$promise$$internal$$invokeCallback 
  (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32.16.01.debug.js:9250:8)
 at lib$es6$promise$$internal$$publish 
  (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:9226:9)
 at lib$es6$promise$asap$$flush 
  (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:9060:8)
```

## Steps to Reproduce, or Live Example

* Link to live example: https://gist.github.com/ndreisg/daaedde3f2547e3e27f6ec0a53c7767d

* Additional details:
  1. Start the sample
  2. Open second Workbook
  3. Wait a few seconds
  4. Select first Workbook again and see the errors in the log

## Your Environment

* Platform: PC desktop
* Host: Excel
* Office version number: tried with various versions
* Operating System: Windows 10

**Comment by RuoyingLiang** [comment]
Thanks for reporting this. But when a new workbook is opened, the selection is moved to the new workbook. There will be no selection in the workbook where add-in is running. You can try that by opening two workbook and select on the second workbook, the selection in the first workbook is disappeared. 

I understand the error may not a good experience and we should think about improve it. But currently you may consider to catch the error. After the selection back to the workbook, the add-in will continue work.

**Comment by RuoyingLiang** [comment]
Could you describe more about your scenario? Thanks!

**Comment by ndreisg** [comment]
Our Add-In [Scan-IT to Office](https://appsource.microsoft.com/de-de/product/office/WA104381026) receives Barcodes and other data from one (or more) mobile devices via a WebSocket connection.
When the Add-In receives data from a device, it inserts it at the current selection and then sets the selection to the next row.
When a user opens a second workbook, the Add-In can't insert the received data anymore.

Maybe I can work around this problem by remembering the last valid selection in the workbooks onSelectionChanged event... I will give it a try :)

**Comment by ndreisg** [comment]
Unfortunately that workaround did not work as expected.
It seems I cannot use the range object across multiple `object.sync` calls.
I get the following error:

`The object path '_reference()' isn't working for what you're trying to do. If you're using the object across multiple "context.sync" calls and outside the sequential execution of a ".run" batch, please use the "context.trackedObjects.add()" and "context.trackedObjects.remove()" methods to manage the object's lifetime.`

I tried to add my object to  `context.trackedObjects` but the error still appears.

See new sample: https://gist.github.com/ndreisg/d88e43b0eeae36e61e6bcfb702f25aed

**Comment by ndreisg** [comment]
I realized that I had to use the ranges `track` method instead of `context.trackedObjects.add()`

Inserting the data now works.

Unfortunately a new problem appeared. After inserting the data I want to select the next row, which doesn't work when another Workbook is focused.

**Comment by ndreisg** [comment]
I finally finished my workaround.
Unfortunately I can't get it to work with Excel API 1.1. 
Since the onSelectionChanged event is not available there, I tried to poll the selection in an interval instead, but it seems `Range.track` and `context.trackedObjects.add` are not working as expected.
Any hints?

**Comment by RuoyingLiang** [comment]
context.trackedObjects.add is a Common API, it's not in Excel API set. Does your add-in work now? could you let me know what's the error you hit?

**Comment by ndreisg** [comment]
Yes my Add-In is now working as long as the `Workbook.onSelectionChanged` event is available (Excel API 1.2 and higher).

In Excel API 1.1 I tried to use an interval instead of the event but I get the following error when I call `mySavedRange.getCell(0,0);` while another Workbook is selected.

```
Error at getSelectedRange()
InvalidObjectPath: The object path 'getSelectedRange()' isn't working for what you're trying to do. If you're using the object across multiple "context.sync" calls and outside the sequential execution of a ".run" batch, please use the "context.trackedObjects.add()" and "context.trackedObjects.remove()" methods to manage the object's lifetime.
 at CommonUtility.validateObjectPath (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:11453:6)
 at ActionFactory.createInstantiateAction (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:12019:4)
 at ClientObject (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:12062:6)
 at Range (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:19671:4)
 at BatchApiHelper.createMethodObject (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:14411:4)
 at Range.prototype.getCell (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:20098:4)
 at Anonymous function (...myScript.js:47:11)
 at Anonymous function (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js:12862:6)
 at r (https://cdn.jsdelivr.net/bluebird/3.5.0/bluebird.min.js:31:9351)
 at i.prototype._settlePromiseFromHandler (https://cdn.jsdelivr.net/bluebird/3.5.0/bluebird.min.js:30:14559)
```
